### PR TITLE
feat: Add Management, RBAC API, RBAC API with Domains tests and fix filePath function removal.

### PR DESCRIPTION
**Pull request by divy9881**
Fix: https://github.com/casbin/casbin-cpp/issues/56

### CODE REVIEW

**Comment by divy9881** [comment]
@xcaptain and @ZipoChan, please review.

**Comment by divy9881** [comment]
This PR also fixes: #56

**Comment by hsluoyz** [comment]
> 
> 
> This PR also fixes: #56

Then you can add `Fix: #xx` into your issue text. GitHub will automatically close it when PR is merged.

File: casbin/enforcer_interface.h [hunk 1]
```diff
@@ -60,10 +60,11 @@ class IEnforcer {
         virtual bool EnforceWithMatcher(string matcher, Scope scope) = 0;
 
         /* RBAC API */
-        virtual vector<string> GetRolesForUser(string name) = 0;
-        virtual vector<string> GetUsersForRole(string name) = 0;
+        virtual vector<string> GetRolesForUser(string name, vector<string> domain) = 0;
```
Comment (hsluoyz) [inline casbin/enforcer_interface.h:63]: Can we make `domain` into an optional arg?

File: casbin/internal_api.cpp [hunk 2]
```diff
@@ -65,8 +65,11 @@ bool Enforcer :: addPolicies(string sec, string p_type, vector<vector<string>> r
         this->BuildIncrementalRoleLinks(policy_add, p_type, rules);
 
     if (this->adapter != NULL && this->auto_save) {
-        void* adapter = this->adapter;
-        ((BatchAdapter *)adapter)->AddPolicies(sec, p_type, rules);
+        try {
+            dynamic_cast<BatchAdapter*>(this->adapter)->AddPolicies(sec, p_type, rules);
+        }
+        catch(UnsupportedOperationException e) {
```
Comment (hsluoyz) [inline casbin/internal_api.cpp:71]: Do we need to do anything in catch?

**Comment by hsluoyz** [comment]
@xcaptain please review.

File: casbin/enforcer_interface.h (hunk 1 above)
Comment (divy9881) [reply casbin/enforcer_interface.h:63]: Yes, we can do that.

File: casbin/internal_api.cpp (hunk 2 above)
Comment (divy9881) [reply casbin/internal_api.cpp:71]: No nothing, just we need to log the message, but casbin-cpp logger has not yet developed, so for now, we leave the catch block empty.

File: casbin/enforcer_interface.h (hunk 1 above)
Comment (hsluoyz) [reply casbin/enforcer_interface.h:63]: I think Golang Casbin makes it optional. Please follow the similar pattern.

File: casbin/internal_api.cpp [hunk 3]
```diff
@@ -116,8 +119,11 @@ bool Enforcer :: removePolicies(string sec, string p_type, vector<vector<string>
         this->BuildIncrementalRoleLinks(policy_add, p_type, rules);
 
     if (this->adapter != NULL && this->auto_save) {
-        void* adapter = this->adapter;
-        ((BatchAdapter *)adapter)->RemovePolicies(sec, p_type, rules);
+        try{
+        dynamic_cast<BatchAdapter*>(this->adapter)->RemovePolicies(sec, p_type, rules);
```
Comment (hsluoyz) [inline casbin/internal_api.cpp:123]: Wrong indentation.

File: casbin/model/assertion.cpp [hunk 4]
```diff
@@ -76,5 +77,5 @@ void Assertion :: BuildRoleLinks(RoleManager* rm) {
 
     // LogUtil :: LogPrint("Role links for: " + Key);
 
-    this->rm->PrintRoles();
+    //this->rm->PrintRoles();
```
Comment (hsluoyz) [inline casbin/model/assertion.cpp:80]: Add a space after //

**Comment by hsluoyz** [comment]
@divy9881 please resolve conflicts.

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**